paper.tex

```diff
@@ -379,6 +379,8 @@ \subsubsection{Adversarial capabilities}
 hundred (or thousand) times from the distribution and computing the mean, such
 an approach is not possible for worst-case robustness.
 
+It is customary for defenses not to impose any computational bounds on an attacker (i.e., the above definitions of adversarial risk consider only the \emph{existence} of adversarial examples, and not the difficulty of finding them). We believe that restricting an adversary's computational power would only be interesting if we could base robustness on a cryptographic assumption (e.g., the model is robust under the assumption that the adversary cannot factor a product of large primes). We are not aware of any defense that achieves this. Stronger restrictions (e.g., the adversary is limited to attacks with $100$ iterations) are typically uninteresting in that they cannot be meaningfully enforced and there is usually no economic reason for an adversary not to spend a little more time to succeed.
```

Comment:
> It is customary for defenses not to impose any computational bounds on an attacker (i.e., the above definitions of adversarial risk consider only the \emph{existence} of adversarial examples

Maksym Andriushchenko found a scenario in which an adversarial defense needed 1000 random restarts per example to bring down the model's accuracy to the fullest extent. Analysis by Synthesis also shows how the adversary budget can swell---it can require 100000 samples per example. On an evaluation dataset of 10000 examples, and with a 100 step attacker, we are left with so many zeros my head is spinning. "Existence" is an appropriate aesthetic in maths where we can _choose_ any element we want without cost, but if there is not a good search procedure for that element, then it is possible for the problem to veer away from empirical study.

Reply:
Similar to my comment above, it is very tricky to say that robustness has really increased here. This is basically saying that the attacker just needs to find a faster attack and the model is easy to break. Unless there's a good reason to believe that optimizing the attack should be hard, this is a pretty weak degree of security, that is also very hard to evaluate or refute. E.g., if the difference between robustness and no robustness is whether the attacker took the time to write down the attack in assembly, run it on TPUs, and come up with a bunch of clever heuristics to squeeze out 1-2 orders of magnitude, that seems like an uninteresting direction to go for.

Reply:
> has _really_ increased here (emphasis mine)
I am not claiming robustness has categorically increased, but I am claiming that computational concerns can be relevant.

> Unless there's a good reason to believe that optimizing the attack should be hard
This could happen, in which case computation is relevant. Of course now it's not currently relevant since attacks are cheap. But if this document is forward-looking, then saying a factor as often important as computation has no relevance seems to exclusively prioritize security precedents at the expense of machine learning precedents. If the document is not trying to be forward-looking, then I absolutely agree computational cost is usually not relevant in evaluation.

Reply:
So I don't think we want to say that computation cost is never relevant.
With the paragraph I added in line 382, I wanted to say that this starts to be relevant once you can formally say that attacking is computationally expensive, or if you can argue that the economic incentives at play make attacking too costly.
To me, this is forward-looking in that we say that right now, computational cost isn't interesting to look at because we really can't say anything meaningful, but if someone writes a paper some day that does make significant progress along these lines, that would be great.

> Rather than making tough calls about the irrelevance of computation, perhaps just recommend increasing the attacker budget in various respects by an order of magnitude to the point of convergence, as the rest of the document currently does? 

So the document already say this for the number of iterations, which is a proxy for the computation time. Did you have something else in mind?

Reply:
> So the document already say this

I acknowledge the document says this for iterations. It should probably do so for the number of random restarts as well.

The text that I took issue with was as follows.

> merely increasing the time it takes an adversary to find an adversarial example... does not constitute an increase in robustness.

Reply:
> merely increasing the time it takes an adversary to find an adversarial example... does not constitute an increase in robustness.

Ah I see, yes stated like this it is a bit too strict. I'll rephrase this to something more along the lines of line 382.

Reply:
I'm glad to see we're not really disagreeing about anything. Thanks!